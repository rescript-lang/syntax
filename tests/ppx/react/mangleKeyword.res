@@jsxConfig({version: 3})

module C0 = {
  @react.component
  let make = (~_open) => React.string(_open)
}
module C1 = {
  @react.component
  external make: (~_open: string) => React.element = "default"
}

let c0 = <C0 _open="x" />
let c1 = <C1 _open="x" />

@@jsxConfig({version: 4, mode: "classic"})

module C0 = {
  @react.component
  let make =
    (@as("open") ~_open, @as("type") ~_type: string) => React.string(_open)
}
module C1 = {
  @react.component
  external make: (@as("open") ~_open: string, @as("type") ~_type: string) => React.element =
    "default"
}

let c0 = <C0 _open="x" _type="t" />
let c1 = <C1 _open="x" _type="t" />

@@jsxConfig({version: 4, mode: "automatic"})

module C0 = {
  @react.component
  let make =
    (@as("open") ~_open, @as("type") ~_type: string) => React.string(_open)
}
module C1 = {
  @react.component
  external make: (@as("open") ~_open: string, @as("type") ~_type: string) => React.element =
    "default"
}

let c0 = <C0 _open="x" _type="t" />
let c1 = <C1 _open="x" _type="t" />
