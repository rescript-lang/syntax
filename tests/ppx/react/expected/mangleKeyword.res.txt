@@jsxConfig({version: 3})

module C30 = {
  @obj external makeProps: (~_open: 'T_open, ~key: string=?, unit) => {"_open": 'T_open} = ""

  @react.component let make = @warning("-16") (~_open) => React.string(_open)
  let make = {
    let \"MangleKeyword$C30" = (\"Props": {"_open": 'T_open}) => make(~_open=\"Props"["_open"])
    \"MangleKeyword$C30"
  }
}
module C31 = {
  @obj external makeProps: (~_open: string, ~key: string=?, unit) => {"_open": string} = ""
  external make: React.componentLike<{"_open": string}, React.element> = "default"
}

let c30 = React.createElement(C30.make, C30.makeProps(~_open="x", ()))
let c31 = React.createElement(C31.make, C31.makeProps(~_open="x", ()))

@@jsxConfig({version: 4, mode: "classic"})

module C4C0 = {
  type props<'T_open, 'T_type> = {@as("open") _open: 'T_open, @as("type") _type: 'T_type}

  @react.component
  let make = ({@as("open") _open, @as("type") _type, _}: props<'T_open, string>) =>
    React.string(_open)
  let make = {
    let \"MangleKeyword$C4C0" = (props: props<_>) => make(props)

    \"MangleKeyword$C4C0"
  }
}
module C4C1 = {
  type props<'T_open, 'T_type> = {@as("open") _open: 'T_open, @as("type") _type: 'T_type}

  external make: @as("open") React.componentLike<props<string, string>, React.element> = "default"
}

let c4c0 = React.createElement(C4C0.make, {_open: "x", _type: "t"})
let c4c1 = React.createElement(C4C1.make, {_open: "x", _type: "t"})

@@jsxConfig({version: 4, mode: "automatic"})

module C4A0 = {
  type props<'T_open, 'T_type> = {@as("open") _open: 'T_open, @as("type") _type: 'T_type}

  @react.component
  let make = ({@as("open") _open, @as("type") _type, _}: props<'T_open, string>) =>
    React.string(_open)
  let make = {
    let \"MangleKeyword$C4A0" = (props: props<_>) => make(props)

    \"MangleKeyword$C4A0"
  }
}
module C4A1 = {
  type props<'T_open, 'T_type> = {@as("open") _open: 'T_open, @as("type") _type: 'T_type}

  external make: @as("open") React.componentLike<props<string, string>, React.element> = "default"
}

let c4a0 = React.jsx(C4A0.make, {_open: "x", _type: "t"})
let c4a1 = React.jsx(C4A1.make, {_open: "x", _type: "t"})
